src/hooks/withQuery.tsx

```diff
@@ -0,0 +1,20 @@
+import { useQuery, UseQueryOptions } from 'react-query'
+import ky from 'ky'
+
+function withQuery<T>(
+  queryKey: string,
+  url: string,
+  options?: UseQueryOptions<T, unknown>
+) {
+  return useQuery<T, unknown>(
+    queryKey,
+    async () => {
+      const response = await ky.get(url)
+      const data: T = await response.json()
+      return data
+    },
+    options
```

Comment:
Here we can add our useful logic to perhaps not run a query unless every value in the key is defined
```suggestion
   { enabled: !Array.isArray(queryKey) || queryKey.every(isPresent), ...(options || {}) }
```